system/mp/mpcomm.cpp

```diff
@@ -1932,16 +1937,29 @@ int CMPConnectThread::run()
         if (sock) {
             try {
                 sock->set_keep_alive(true);
-                size32_t rd;
+                size32_t rd = 0;
                 SocketEndpoint _remoteep;
                 SocketEndpoint hostep;
                 SocketEndpointV4 id[2];
                 traceSlowReadTms("MP: initial accept packet from", sock, &id[0], sizeof(id), sizeof(id), rd, CONFIRM_TIMEOUT, CONFIRM_TIMEOUT_INTERVAL);
-                if (rd != sizeof(id))
+
+                if (0 == rd)
                 {
-                    StringBuffer errMsg("MP Connect Thread: invalid number of connection bytes serialized from ");
+                    if (mpTraceLevel > 1)
+                    {
+                        // cannot get peer addresss as socket state is now ss_shutdown (unless we want to allow this in getPeerEndpoint())
```

Comment:
minor: doesn't matter, as we won't be tracing (by default) anyway, but it would have probably been okay to retrieve peer endpoint unconditionally (i.e. move out of _FULLTRACE above).